### PR TITLE
Need to block access to kernel file systems in /proc and /sys

### PR DESCRIPTION
**Pull request by rhatdan**
Users of kpod run could use these file systems to perform a breakout
or to learn valuable system information.

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@baude @umohnani8 @mheon PTAL

**Comment by mheon** [comment]
LGTM, but you're getting test failures on Cent 7.

**Comment by umohnani8** [comment]
LGTM

**Comment by mheon** [comment]
@rh-atomic-bot r+

File: cmd/kpod/spec.go [hunk 1]
```diff
@@ -17,6 +17,33 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+func blockAccessToKernelFilesystems(config *createConfig, g *generate.Generator) {
+	if !config.privileged {
+		for _, mp := range []string{
+			"/proc/kcore",
+			"/proc/latency_stats",
```
Comment (TomSweeneyRedHat) [inline cmd/kpod/spec.go]: latency_stats isn't present in /proc on my RHEL PC.  If it's not present on what ever machine this is being run on, I think that's OK but just wanted to verify.

File: cmd/kpod/spec.go [hunk 2]
```diff
@@ -17,6 +17,33 @@ import (
...
+			g.AddLinuxMaskedPaths(mp)
+		}
+
+		for _, rp := range []string{
+			"/proc/asound",
+			"/proc/bus",
+			"/proc/fs",
+			"/proc/irq",
+			"/proc/sys",
+			"/proc/sysrq-trigger",
```
Comment (TomSweeneyRedHat) [inline cmd/kpod/spec.go]: should sqsrq-trigger be read-only?  It seems to have only write perms for the root user on my machines.

File: cmd/kpod/spec.go (hunk 1 above)
Comment (rhatdan) [reply cmd/kpod/spec.go]: Yup this is attempting to hide them, o them not existing is fine.

File: cmd/kpod/spec.go (hunk 2 above)
Comment (rhatdan) [reply cmd/kpod/spec.go]: Not sure, this code was taken from CRI-O

**Pull request closed without merging**